src/Ext/Techno/Hooks.cpp

```diff
@@ -543,3 +543,62 @@ DEFINE_HOOK(0x70EFE0, TechnoClass_GetMaxSpeed, 0x6)
 	return SkipGameCode;
 }
 
+DEFINE_HOOK(0x5F4032, ObjectClass_FallingDown_ToDead, 0x6)
+{
+	GET(ObjectClass*, pThis, ESI);
+
+	if (auto const pTechno = abstract_cast<TechnoClass*>(pThis))
+	{
+		auto pCell = pTechno->GetCell();
+
+		if (!pCell ||
+			!pCell->IsClearToMove(pTechno->GetTechnoType()->SpeedType,
+			true, true, -1, pTechno->GetTechnoType()->MovementZone,
+				pCell->GetLevel(), pCell->ContainsBridge()))
+			return 0;
+
+		if (auto const pTypeExt = TechnoTypeExt::ExtMap.Find(pTechno->GetTechnoType()))
+		{
+			double ratio = 0.0;
+
+			if (pCell->Tile_Is_Water() && !pTechno->OnBridge)
+				ratio = pTypeExt->FallingDownDamage_Water.Get(pTypeExt->FallingDownDamage.Get());
+			else
+				ratio = pTypeExt->FallingDownDamage.Get();
+
+			int pDamage = 0;
+
+			if (ratio < 0.0)
+				pDamage = int(pThis->Health * abs(ratio));
+			else if (ratio >= 0.0 && ratio <= 1.0)
+				pDamage = int(pThis->GetTechnoType()->Strength * ratio);
+			else
+				pDamage = int(ratio);
+
+			pThis->ReceiveDamage(&pDamage, 0, RulesClass::Instance->C4Warhead, nullptr, true, true, nullptr);
+
+			if (pThis->Health > 0 && pThis->IsAlive)
+			{
+				pThis->IsABomb = false;
+
+				if (pThis->WhatAmI() == AbstractType::Infantry)
+				{
+					auto pInf = abstract_cast<InfantryClass*>(pTechno);
+
+					if (pCell->Tile_Is_Water() && pInf->SequenceAnim != Sequence::Swim)
+						pInf->PlayAnim(Sequence::Swim, true, false);
+					else if (!pCell->Tile_Is_Water() && pInf->SequenceAnim != Sequence::Guard)
+						pInf->PlayAnim(Sequence::Guard, true, false);
```

Comment:
const auto pInf = static_cast<InfantryClass*>(pTechno);

				if (pCell->Tile_Is_Water())
				{
					if (pInf->SequenceAnim != Sequence::Swim)
						pInf->PlayAnim(Sequence::Swim, true, false);
				}
				else if (pInf->SequenceAnim != Sequence::Guard)
				{
					pInf->PlayAnim(Sequence::Guard, true, false);
				}
```suggestion
				const auto pInf = static_cast<InfantryClass*>(pTechno);

				if (pCell->Tile_Is_Water())
				{
					if (pInf->SequenceAnim != Sequence::Swim)
						pInf->PlayAnim(Sequence::Swim, true, false);
				}
				else if (pInf->SequenceAnim != Sequence::Guard)
				{
					pInf->PlayAnim(Sequence::Guard, true, false);
				}
```